src/module/chat.ts

```diff
@@ -66,14 +107,111 @@ export const createChatData = async (templateData: TemplateData, roll?: Roll) =>
     if (roll) {
         chatData['sound'] = CONFIG.sounds.dice;
     }
-    const rollMode = templateData.rollMode ?? game.settings.get('core', 'rollMode');
+    const rollMode = templateData.rollMode ?? game.settings.get(CORE_NAME, CORE_FLAGS.RollMode);
 
-    if (['gmroll', 'blindroll'].includes(rollMode)) chatData['whisper'] = ChatMessage.getWhisperRecipients('GM');
+    if (['gmroll', 'blindroll'].includes(rollMode as string)) chatData['whisper'] = ChatMessage.getWhisperRecipients('GM');
     if (rollMode === 'blindroll') chatData['blind'] = true;
 
     return chatData;
 };
 
+
+
+export async function ifConfiguredCreateDefaultChatMessage(roll: ShadowrunRoll, {actor, title, rollMode}) {
+    if (game.settings.get(SYSTEM_NAME, FLAGS.DisplayDefaultRollCard)) {
+        await roll.toMessage({
+            speaker: ChatMessage.getSpeaker({ actor: actor }),
+            flavor: title,
+            rollMode: rollMode,
+        });
+    }
+}
+
+export async function createItemChatMessage(options: ItemChatMessageOptions) {
+    const templateData = createChatTemplateData(options);
+    const chatData = await createChatData(templateData);
+    return await ChatMessage.create(chatData);
+}
+
+function createChatTemplateData(options: ItemChatMessageOptions): ItemChatTemplateData {
+    // field extraction is explicit to enforce visible data flow to ensure clean data.
+    // NOTE: As soon as clear data dynamic data flow can be established, this should be removed for a simple {...options}
+    let {actor, item, description, tests, header} = options;
+
+    return {
+        actor,
+        item,
+        header,
+        description,
+        tests
+    }
+}
+
+export async function createRollChatMessage(roll: ShadowrunRoll, options: RollChatMessageOptions): Promise<Entity<any>> {
+    const templateData = getRollChatTemplateData(roll, options);
+    const chatData = await createChatData(templateData, roll);
+    // TODO: What does displaySheet even do?
```

Comment:
I think I copied it from some other example... from looking at the source code I don't think it has any effect really for a chat message. https://foundryvtt.com/api/foundry.js.html#line28273

Reply:
![grafik](https://user-images.githubusercontent.com/146255/96563751-9d7b7000-12c2-11eb-8588-f2897dac2777.png)

Thanks for the insight.

Reply:
It's gone now, couldn't find any resulting behaviour changes. Very agile.